src/components/Navbar.tsx

```diff
@@ -56,26 +63,32 @@ function Navbar() {
                 </div>
             </div>
             <div className="navbar-end">
-                <button
-                    id="button-search"
-                    title="Search"
-                    className="btn btn-ghost btn-circle"
-                >
-                    <svg
-                        xmlns="http://www.w3.org/2000/svg"
-                        className="h-5 w-5"
-                        fill="none"
-                        viewBox="0 0 24 24"
-                        stroke="currentColor"
-                    >
-                        <path
-                            strokeLinecap="round"
-                            strokeLinejoin="round"
-                            strokeWidth="2"
-                            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
-                        />
-                    </svg>
-                </button>
+                {search ? (
+                    <SearchBar setSearchResult={setSearchResult!} />
```

Comment:
Nice prop drilling going on here. Maybe time for some state management?

Reply:
Even if that's just context and useReduce

Reply:
I fixed this by removing it and using zustand to store search query.